delphix/mongo-plugin
# CE-68 MongoDB Plugin: Capture Mongo error from MongoDB logs

### Problem

Capture mongod command line logs and display it in the debug.log as well as relay them to UI.



### Solution

Capture the logs from the folloing two sources of mongod logs on stg/tgt.
1. conf file
2. logpath specified in the command

In order to do so, we are executing an additional command on stg/tgt host to fetch and parse any erros from mongod logpath.




### Testing Done

Following test-cases(negative) 
1. mongod command line with --logpath option which fails eventually
Steps:-
- while creating vdb config, specify the already running port
- vdb creation would fail.
- expected results: Following lines are getting displayed in UI, where second line is the actual mongod error.

An error of type "Exception" was detected while running the plugin operation "Virtual Source Configure": about to fork child process, waiting until server is ready for connections.forked process: 24834ERROR: child process failed, exited with 48To see additional information in this output, start without the "--fork" option.

{'code': 9001, 'codeName': 'SocketException', 'errmsg': 'Address already in use'}


2. mongod command line without --logpath option but with " -f " conf file option, which fails eventually.
Steps:-
- create seed_vdb_test
- stop vdb from UI
- mv /mnt/provision/seed_vdb_test/s0m0  /mnt/provision/seed_vdb_test/s0m0.bkp
- upload the plugin with fix
- start vdp
- it should throw error.
- expected results: Following lines are getting displayed in UI, where second line is the actual mongod error.

Exception: about to fork child process, waiting until server is ready for connections.forked process: 21792ERROR: child process failed, exited with 100To see additional information in this output, start without the "--fork" option.
NonExistentPath: Data directory /mnt/provision/seed_vdb_test/s0m0 not found. Create the missing directory or specify another path using (1) the --dbpath command line option, or (2) by adding the 'storage.dbPath' option in the configuration file.

(Screenshots of the UI errors are attached to this PR)
